File: system-test/dotnet/Dsl.Core/UseCase/Shop/UseCases/GoToShop.cs
using Driver.Port.Shop;
using Dsl.Core.Shop.UseCases.Base;
using Driver.Port.Shop.Dtos.Error;
using Common;
using Dsl.Core.Shared;

namespace Dsl.Core.Shop.UseCases;

public class GoToShop : BaseShopCommand<VoidValue, VoidVerification>
{
    public GoToShop(IShopDriver driver, UseCaseContext context)
        : base(driver, context)
    {
    }

    public override async Task<ShopUseCaseResult<VoidValue, VoidVerification>> Execute()
    {
        var result = await _driver.GoToShopAsync();

        return new ShopUseCaseResult<VoidValue, VoidVerification>(
            result,
            _context,
            (response, ctx) => new VoidVerification(response, ctx));
    }
}

File: system-test/java/src/main/java/com/optivem/shop/dsl/core/usecase/shop/usecases/GoToShop.java
package com.optivem.shop.dsl.core.usecase.shop.usecases;

import com.optivem.shop.dsl.driver.port.shop.ShopDriver;
import com.optivem.shop.dsl.core.usecase.shop.usecases.base.BaseShopUseCase;
import com.optivem.shop.dsl.core.shared.UseCaseResult;
import com.optivem.shop.dsl.core.shared.UseCaseContext;
import com.optivem.shop.dsl.core.shared.VoidVerification;

public class GoToShop extends BaseShopUseCase<Void, VoidVerification> {
    public GoToShop(ShopDriver driver, UseCaseContext context) {
        super(driver, context);
    }

    @Override
    public UseCaseResult<Void, VoidVerification> execute() {
        var result = driver.goToShop();
        return new UseCaseResult<>(result, context, VoidVerification::new);
    }
}



